pkg/api/controller.go

```diff
@@ -72,6 +71,10 @@ type actionsHandler interface {
 	ListRunTaskResults(ctx context.Context, repositoryID string, runID string, after string) (actions.TaskResultIterator, error)
 }
 
+type UploadPathProvider interface {
```

Comment:
Another interface definition?

Reply:
Yes - prefer to have local requirement as interface by the package that consume the implementation than having a global one.